lbrynet/lbrynet_daemon/LBRYDaemonServer.py

```diff
@@ -179,12 +361,35 @@ def render_GET(self, request):
     #     call.addErrback(lambda err: log.info("Error: " + str(err)))
     #     call.cancel()
 
+class LBRYFileUpload(resource.Resource):
+    """
+    Accepts a file sent via the file upload widget in the web UI, saves
+    it into a temporary dir, and responds with a JSON string containing
+    the path of the newly created file.
+    """
+
+    def __init__(self, api):
+        self._api = api
+
+    def render_POST(self, request):
+        origfilename = request.args['file_filename'][0]
+        uploaded_file = request.args['file'][0]  # Temp file created by request
+
+        # Move to a new temporary dir and restore the original file name
```

Comment:
In this case, the "what" and the "why" are kind of the same--the file starts in a tempfile with a randomized name and we need to put it in a new spot with the correct name.

Also, `move_to_temp_dir_and_restore_file_name`  is the shortest function name I can think of that would convey the needed info on its own, and that's pretty long winded. I guess I'll just modify LBRYDaemonRequest so it directly puts them in the right place (makes the code even less reusable since we're creating temp dirs as well as temp files, but whatever).